File: MyApplication2/app/src/main/java/com/murach/myapplication/AllActivity/MainMenu.java
package com.murach.myapplication.AllActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import com.murach.myapplication.WithOffline.MainActivity;
import com.murach.myapplication.R;

public class MainMenu extends AppCompatActivity implements View.OnClickListener {
    Button play_offline_button, play_online_button, match_history_button, settings_button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_menu);
        getButton();
        play_offline_button.setOnClickListener(this);
        match_history_button.setOnClickListener(this);
        settings_button.setOnClickListener(this);
    }

    private void getButton() {
        play_offline_button = findViewById(R.id.play_offline_button);
        match_history_button = findViewById(R.id.match_history_button);
        settings_button = findViewById(R.id.settings_button);
    }

    @Override
    public void onClick(View view) {
        int id = view.getId();
        if (id == R.id.play_offline_button) {
            Intent intent = new Intent(this, MainActivity.class);
            startActivity(intent);
        }


        if (id == R.id.match_history_button) {
            Intent intent = new Intent(this, MatchHistory.class);
            startActivity(intent);
        }

        if (id == R.id.settings_button) {
            Intent intent = new Intent(this, ChessSettings.class);
            startActivity(intent);
        }
    }
}

File: MyApplication2/app/src/main/java/com/murach/myapplication/AllActivity/MainMenu.kt
package com.murach.myapplication.AllActivity

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.murach.myapplication.R
import com.murach.myapplication.WithComputer.BotActivity

import com.murach.myapplication.WithOffline.MainActivity
import com.murach.myapplication.WithWorld.WorldGamePlayActivity

class MainMenu : AppCompatActivity(), View.OnClickListener {
    private lateinit var playOfflineButton: Button
    private lateinit var playOnlineButton: Button
    private lateinit var matchHistoryButton: Button
    private lateinit var settingsButton: Button
    private lateinit var playComputerButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_menu)
        getButton()
        playOfflineButton.setOnClickListener(this)
        matchHistoryButton.setOnClickListener(this)
        settingsButton.setOnClickListener(this)
        playOnlineButton.setOnClickListener(this)
        playComputerButton.setOnClickListener(this)
    }

    private fun getButton() {
        playOfflineButton = findViewById(R.id.play_offline_button)
        matchHistoryButton = findViewById(R.id.match_history_button)
        settingsButton = findViewById(R.id.settings_button)
        playOnlineButton = findViewById(R.id.play_online_button)
        playComputerButton = findViewById(R.id.play_computer_button)
    }

    override fun onClick(view: View) {
        val id = view.id
        when (id) {
            R.id.play_offline_button -> {
                val intent = Intent(this, MainActivity::class.java)
                startActivity(intent)
            }
            R.id.match_history_button -> {
                val intent = Intent(this, ChatActivity::class.java)
                startActivity(intent)
            }
            R.id.settings_button -> {
                val intent = Intent(this, ChessSettings::class.java)
                startActivity(intent)
            }
            R.id.play_online_button -> {
                val intent = Intent(this, WorldGamePlayActivity::class.java)
                startActivity(intent)
            }
            R.id.play_computer_button -> {
                val intent = Intent(this, BotActivity::class.java)
                startActivity(intent)
            }
        }
    }
}
